src/Doctrine/Orm/Metadata/Resource/DoctrineOrmLinkFactory.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Metadata\Resource;
+
+use ApiPlatform\Api\ResourceClassResolverInterface;
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Link;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use ApiPlatform\Metadata\Resource\Factory\LinkFactoryInterface;
+use ApiPlatform\Metadata\Resource\Factory\PropertyLinkFactoryInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+
+/**
+ * @internal
+ */
+final class DoctrineOrmLinkFactory implements LinkFactoryInterface, PropertyLinkFactoryInterface
+{
+    public function __construct(private readonly ManagerRegistry $managerRegistry, private readonly PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, private readonly ResourceClassResolverInterface $resourceClassResolver, private readonly LinkFactoryInterface&PropertyLinkFactoryInterface $linkFactory)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createLinkFromProperty(ApiResource|Operation $operation, string $property): Link
+    {
+        return $this->linkFactory->createLinkFromProperty($operation, $property);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createLinksFromIdentifiers(ApiResource|Operation $operation): array
+    {
+        return $this->linkFactory->createLinksFromIdentifiers($operation);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createLinksFromRelations(ApiResource|Operation $operation): array
+    {
+        $links = $this->linkFactory->createLinksFromRelations($operation);
+
+        $resourceClass = $operation->getClass();
+        if (!($manager = $this->managerRegistry->getManagerForClass($resourceClass)) instanceof EntityManagerInterface) {
+            return $links;
+        }
+
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $property) {
+            $doctrineMetadata = $manager->getClassMetadata($resourceClass);
+            if (!$doctrineMetadata->hasAssociation($property)) {
+                continue;
+            }
+
+            $relationClass = $doctrineMetadata->getAssociationTargetClass($property);
+            if (!($mappedBy = $doctrineMetadata->getAssociationMappedByTargetField($property)) || !$this->resourceClassResolver->isResourceClass($relationClass)) {
+                continue;
+            }
+
+            $link = (new Link())->withFromProperty($property)->withToProperty($mappedBy)->withFromClass($resourceClass)->withToClass($relationClass);
```

Comment:
what about using named arguments here? (less method calls)

Reply:
Sure, why not.